group/forms.py

```diff
@@ -15,32 +15,65 @@
     AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
     LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
     OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-    SOFTWARE.
-    '''
+    SOFTWARE.'''
 
 from django import forms
-from group.models import Group
+from django.db.models import Q
+from group.models import Group, Announce
+from users.models import User
 
 class GroupForm(forms.ModelForm):
+    def __init__(self, *args, **kwargs):
+        super(GroupForm, self).__init__(*args, **kwargs)
+        # access object through self.instance...
+        self.fields['coowner'].queryset = User.objects.exclude(user_level=User.USER)
+
     class Meta:
         model = Group
         fields = [
             'gname',
-            'owner',
             'coowner',
-            'member',
+            #'member',
             'description',
-            'announce',
             'trace_contest',
         ]
 
 class GroupFormEdit(forms.ModelForm):
+    def __init__(self, *args, **kwargs):
+        super(GroupFormEdit, self).__init__(*args, **kwargs)
+        # access object through self.instance...
+        self.fields['coowner'].queryset = User.objects.exclude(user_level=User.USER)
+        #self.fields['member'].queryset = User.objects.exclude(self.fields['coowner'].queryset)
     class Meta:
         model = Group
         fields = [
+            'gname',
             'coowner',
-            'member',
+            #'member',
             'description',
-            'announce',
             'trace_contest',
         ]
+
+class Coowner_GroupFormEdit(forms.ModelForm):
```

Comment:
I think this form is not necessary. Coowner list can be hide by other way. You can ask @Changron how to do.